src/casync-http.c

```diff
@@ -516,24 +550,26 @@ int main(int argc, char* argv[]) {
                 .sa_flags = SA_RESTART,
         };
 
-        int r;
+        int r, i;
 
         assert_se(sigaction(SIGPIPE, &sa, NULL) >= 0);
 
         r = parse_argv(argc, argv);
         if (r <= 0)
                 goto finish;
 
-        if (argc < 2) {
+        i = r;
+
+        if (argc < i) {
```

Comment:
hmm, instead of propagating this from parse_argv() i think we should compare to optind directly here...